src/main/java/org/codehaus/mojo/clirr/ClirrReport.java

```diff
@@ -340,9 +340,19 @@ private ResourceBundle getBundle( Locale locale )
 
     public boolean canGenerateReport()
     {
+        boolean generate = true;
         try
         {
-            return canGenerate();
+            if ( skip )
+            {
+                getLog().info( "Skipping execution" );
+                generate = false;
```

Comment:
why not a simple return false?
no need to use var and so many lines.
if (skip)
{
  getLog().info( "Skipping execution" );
  return false
}
return canGenerate();
IMHO it's more simple/readable (but maybe it's only me :-) )